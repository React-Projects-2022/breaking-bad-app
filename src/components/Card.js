import PropTypes from "prop-types";
import { Button } from ".";
import { openExternalURL } from "../helpers/navigate";

const Card = ({ character }) => {
  const goToMoreDetails = (url) => openExternalURL(url);
  return (
    <div className="col" key={character.char_id}>
      <div className="card">
        <div className="image-section">
          <img
            src={character.img}
            className="card-img-top"
            alt={character.name}
          />
          <span className="bottom-left">{character.portrayed}</span>
        </div>
        <div className="card-body">
          <h5 className="card-title">{character.name}</h5>
          {
            (character.char_id === undefined) ?
            (
              <>
              <p className="card-text truncate">{character.description}</p>
              
              <Button
                key={`Wiki_${character.id}`}
                label={"Más información"}
                param={character.url}
                action={goToMoreDetails}
              />
              </>
            ): (
              <>
              <p className="card-text">{character.occupation.join(", ")}</p>
              <p className="card-text">
                <small className="text-muted">
                  Appearance Seasons: {character.appearance.join(", ")}
                </small>
              </p>
              </>
            )
          }
          
        </div>
      </div>
    </div>
  )
};

// Para obligar pasar el argumento para el objeto
Card.propTypes = {
  character: PropTypes.object.isRequired,
};
export default Card;
