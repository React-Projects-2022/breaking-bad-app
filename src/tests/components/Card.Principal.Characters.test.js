import Card from "../../components/Card";
import { shallow } from "enzyme";

describe("Pruebas de <Card /> en 'Principal Characters'", () => {
  const character = {
    id: "1",
    name: "Walter White",
    description:
      'Walter "Walt" Hartwell White, también conocido por su seudónimo clandestino "Heisenberg", es un químico y profesor de química de Albuquerque, Nuevo México que, tras ser diagnosticado con cáncer de pulmón inoperable, decide empezar a fabricar metanfetamina para poder mantener a su familia provista de cara al futuro"',
    portrayed: "Bryan Cranston",
    total_episodes: 62,
    votes: 3797,
    url: "https://es.wikipedia.org/wiki/Walter_White",
    img: "https://raw.githubusercontent.com/graphql-course/5-breaking-bad-graphql-voting/master/photos/BB-S5B-Walt-590.jpg",
    __typename: "Character",
  };
  const wrapper = shallow(
    <Card character={character} useLabel={"More info"} />
  );

  test("Debe de mostrar <Button /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Comprobar que las propiedades de imagen correctas", () => {
    const imageProps = wrapper.find("img").props();
    expect(imageProps.src).toBe(character.img);
    expect(imageProps.alt).toBe(character.name);
    expect(imageProps.className).toBe("card-img-top");
  });

  test("Comprobar que texto del nombre de personaje es correcto", () => {
    expect(wrapper.find("h5").text()).toBe(character.name);
  });

  test("Comprobar que la descripción es la correcta", () => {
    expect(wrapper.find(".truncate").at(0).text()).toBe(character.description);
  });
  /*
  test("Comprobar que las temporadas en que aparece se renderizan bien en el formato deseado", () => {
    expect(wrapper.find("p").at(1).text()).toBe(
      `Appearance Seasons: ${character.appearance.join(", ")}`
    );
  });*/
});
