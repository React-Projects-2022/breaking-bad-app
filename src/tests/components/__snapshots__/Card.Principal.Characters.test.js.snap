// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <Card /> en 'Principal Characters' Debe de mostrar <Button /> correctamente 1`] = `
<div
  className="col"
>
  <div
    className="card"
  >
    <div
      className="image-section"
    >
      <ImageRenderer
        customClass="card-img-top"
        height={470}
        thumb="https://raw.githubusercontent.com/graphql-course/5-breaking-bad-graphql-voting/master/photos/BB-S5B-Walt-590.jpg"
        title="Walter White"
        url="https://raw.githubusercontent.com/graphql-course/5-breaking-bad-graphql-voting/master/photos/BB-S5B-Walt-590.jpg"
        width={600}
      />
      <span
        className="bottom-left"
      >
        Bryan Cranston
      </span>
    </div>
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Walter White
      </h5>
      <p
        className="card-text truncate"
      >
        Walter "Walt" Hartwell White, también conocido por su seudónimo clandestino "Heisenberg", es un químico y profesor de química de Albuquerque, Nuevo México que, tras ser diagnosticado con cáncer de pulmón inoperable, decide empezar a fabricar metanfetamina para poder mantener a su familia provista de cara al futuro"
      </p>
      <Button
        action={[Function]}
        key="Wiki_1"
        label="More info"
        param="https://es.wikipedia.org/wiki/Walter_White"
      />
    </div>
  </div>
</div>
`;
